### PR TITLE
Fix #8579 Review showElevation property of MousePosition plugin

### LINKED ISSUE #8579: Review showElevation property of MousePosition plugin
## Description

We introduced the concept of `terrain` layer for cesium here https://github.com/geosolutions-it/MapStore2/issues/8101 but this new layer could not used by the `showElevation` property of MousePosition because leaflet and openlayers have not the `terrain` type implemented. 

We need to investigate on following:

- evaluate to separate terrain concept from mouse elevation concept, moving the elevation layer configuration to the MousePosition plugin
- (optional) evaluate the refactor of mouse position from the mouse move event of the core map to a map support. We could technically move all the logic related to the event listener to for mouse position to the plugin if it used only there

**What kind of improvement you want to add?** (check one with "x", remove the others)
- [x] Minor changes to existing features

## Other useful information

This could be solved by using the `wms` type with the `useForElevation` property equal true in the current implementation. We kept support of these while implementing the new terrain type.

Connected to https://github.com/geosolutions-it/MapStore2/issues/8067

### PR DESCRIPTION
**Pull request by allyoucanmap**
## Description

This PR introduces a new type of layer called `elevation` to get height information to display in the mouse position plugin. This split the concept of terrain for visualization and elevation as data information.

In particular this note included in the documentation:

> Note that in the Cesium 3D viewer all the heights are relative to the WGS84 ellipsoid while usually locally DTM/DEM file could have an elevation value relative to the mean sea level (MSL). So with this new layer it's possible to have a terrain layer with the correct WGS84 ellipsoidal height while querying the mouse position with a MSL height.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x]  Minor changes to existing features

## Issue

**What is the current behavior?**

#8579

**What is the new behavior?**

Introduction of a new layer of type elevation

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [x] No

### CODE REVIEW

File: docs/developer-guide/mapstore-migration-guide.md [hunk 1]
```diff
@@ -22,6 +22,45 @@ This is a list of things to check if you want to update from a previous version
 
 ## Migration from 2023.02.xx to 2024.01.00
 
+### Using `elevation` layer type instead of wms layer with useForElevation property
+
+The wms layer with useForElevation property is deprecated and a `elevation` layer introduced in substitution.
```
Comment (offtherailz) [inline docs/developer-guide/mapstore-migration-guide.md:27]:
```suggestion
The wms layer with `useForElevation` property is deprecated and a `elevation` layer introduced in substitution.
```

**Review by offtherailz — CHANGES_REQUESTED** [review]
I updated the documentation to make it more understandable. Please take a look at my changes.

I wasn't able to test it: 

image

Context used: 
[additionalLayers-new-context.json](https://github.com/geosolutions-it/MapStore2/files/13718199/additionalLayers-new-context.json)

Did I missed something?

**Comment by allyoucanmap** [comment]
@offtherailz here a working context: [elevation-layer-context.json](https://github.com/geosolutions-it/MapStore2/files/13739247/elevation-layer-context.json)

**Review by offtherailz — APPROVED** [review]

**Comment by offtherailz** [comment]
@ElenaGallo, could you please test this on  DEV (as well as deployed) ? Thank you

**Pull request merged**